universal-application-tool-0.0.1/app/auth/oidc/OidcCiviFormProfileAdapter.java

```diff
@@ -53,11 +52,37 @@ public OidcCiviFormProfileAdapter(
 
   protected abstract void adaptForRole(CiviFormProfile profile, ImmutableSet<Roles> roles);
 
+  protected Optional<String> getAuthorityId(OidcProfile oidcProfile) {
+    // In OIDC the user is uniquely identified by the iss(user) and sub(ject) claims.
+    // https://openid.net/specs/openid-connect-core-1_0.html#IDToken
+    //
+    // We combine the two to create the unique authority id.
+    // Issuer is necessary as CiviForm has different authentication systems for Admins and
+    // Applicants.
+    String issuer = oidcProfile.getAttribute("iss", String.class);
+    // Subject identifies the specific user in the issuer.
+    // Pac4j treats the subject as special, and you can't simply ask for the "sub" claim.
+    String subject = oidcProfile.getId();
+    if (issuer == null || subject == null) {
+      return Optional.empty();
+    }
+    // This string format can never change. It is the unique ID for OIDC based account.
+    return Optional.of(String.format("iss: %s sub: %s", issuer, subject));
```

Comment:
thought (non-blocking): if we ever have need to search for a particular ID by sub or issuer in the DB, it'd be slightly easier if the hard-coded portions of the string weren't in here.

Reply:
I think that scenario is rough eitherway.

Since we're constructing a "meta" value your scenario would only care about part of the stored value, so we'd need partial search ability

without the field names in the value it then is a tad harder to interpret.

Perhaps this should be a json string... but not sure I want to rely on deterministic stringification of a json object.

Open to more discussion here since there is a point of no return.

Reply:
Likewise, SAML will have different constituent parts.. so a general search would seemingly have to be either a full substring search index or tokenized index and full token matches.